src/main/java/codeu/controller/FileUploadServlet.java

```diff
@@ -100,16 +128,34 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
 	    if(inputStream!=null) {
 	      BufferedImage content = ImageIO.read(inputStream);
 	      String contentType = filePart.getContentType();
+	      UUID conversationId = null;
+	      /* If this is uploaded from a conversation, get the conversation id. */
+	      if(uri.equals("/chat/FileUploadServlet")) {
+	      	Conversation conversation = conversationStore.getConversationWithTitle(chatName);
+	      	conversationId = conversation.getId();
+	      }
 	      Media media = new Media(UUID.randomUUID(), user.getId(), 
-	        mediaTitle, Instant.now(), content, contentType);
+	        mediaTitle, Instant.now(), content, contentType, conversationId);
 	      if(mediaStore.getProfilePicture(user.getName())!=null) {
 	        mediaStore.getProfilePicture(user.getName()).setIsProfilePicture(false);
 	      }
-	      mediaStore.setProfilePicture(user.getName(), media);
+	      if(uri.equals("/profile/FileUploadServlet")) {
+	      	mediaStore.setProfilePicture(user.getName(), media);
+	      }
+	      if(uri.equals("/chat/FileUploadServlet")) {
```

Comment:
nitpick:
You have multiple
if(uri.equals("/chat/FileUploadServlet")) { } 
blocks.
It is preferred to (for readability and easy debugging) to combine
these into a single block